test/Datasource.test.js

```diff
@@ -57,10 +62,55 @@ describe('Test all datasouses from ../src/lib/datasouce', () => {
         });
     
         it('should resolve with null', ()=> {
-          return dataSource.delete().then((res) => {
-            expect(res).toEqual(null);  
+          return dataSource.delete('key').then((res) => {
+            expect(res).toEqual(true);  
           });
         });
       })
 
+      describe('#chaining', () => {
+        it('get shoud return null', () => {
+          return emptyDataSource.get('key').then((res) => {
+            expect(res).toEqual(null);  
+          });
+        });
+
+        it('set>get shoud return with seted value', () => {
+          return emptyDataSource.set('key', 'value')
+            .then(() =>  emptyDataSource.get('key').then())
+            .then((res) => {
+              expect(res).toEqual('value'); 
+            });
+        });
+
+        it('set>get>delete shoud return with true', () => {
+          return emptyDataSource.set('key', 'value')
+            .then(() =>  emptyDataSource.get('key').then())
```

Comment:
Review pls this(and every next) chaining .then(), couse I think, it can be some more easy.

Reply:
True, you do not actually need to call this empthy then function every time:
`emptyDataSource.get('key').then()`, it is the same as 
- Do Function
- Return Promise
- Then do nothing
- Return Promise

2 first steps are quite enough =)

Reply:
resolved